### ISSUE TITLE
RFC convertor

### ISSUE THREAD

**Issue by KOLANICH** [kaitai-io/kaitai_struct#30]
RFC by IETF often have descriptions of data structures as ascii tables, for example https://tools.ietf.org/html/rfc4271#section-4.2 . It'd be nice to have a tool to convert such descriptions to descriptions available for such tools as KS.

**Comment by GreyCat** [comment]
The main problem here is that RFCs have dozens of different variantions of binary format descriptions. You've already provided one example. Here are a few others, somewhat or completely different:

* https://tools.ietf.org/html/rfc2083#page-22
* https://tools.ietf.org/html/rfc821#page-4
* https://tools.ietf.org/html/rfc3784#page-3

etc.

**Comment by KOLANICH** [comment]
Yes, there are several variations. At first it'd be nice to implement the ones which are table based, because
* they are only for binary structures, there is no sense to describe text protocol like smtp as a binary structures;
* They always have the same format, the variations are minor: the length and offset of fields are determined only by their borders placement in the text and by the reference frame provided by the rulers on top.

**Comment by GreyCat** [comment]
Ok, that's definitely possible. Probably developing it as a distinct converter script is a way to go - given that format is pretty simple, it shouldn't be that hard. Feel free to contribute it if you feel like it, or maybe someone else would be interested in taking on this task.

**Comment by KOLANICH** [comment]
https://github.com/mitls/quackyducky - contains some kind of rfc parser

**Comment by arekbulski** [comment]
Are those not compiled out of some latex sources?

**Comment by KOLANICH** [comment]
As I remember, they are not.
